doc/OnlineDocs/modeling_extensions/gdp/modeling.rst

```diff
@@ -0,0 +1,415 @@
+| Exclusive OR | :code:`Y[1] ^ Y[2]`    | :code:`Y[1].xor(Y[2])`            | :code:`Xor(Y[1], Y[2])`        |
++--------------+------------------------+-----------------------------------+--------------------------------+
+| Implication  | | :code:`Y[1] >> Y[2]` | :code:`Y[1].implies(Y[2])`        | :code:`Implies(Y[1], Y[2])`    |
+|              | | :code:`Y[2] << Y[1]` |                                   |                                |
++--------------+------------------------+-----------------------------------+--------------------------------+
+| Equivalence  | :code:`Y[1] == Y[2]`   | :code:`Y[1].equivalent_to(Y[2])`  | :code:`Equivalent(Y[1], Y[2])` |
++--------------+------------------------+-----------------------------------+--------------------------------+
+
+In addition, the following constraint programming inspired operators are provided: ``Exactly``, ``AtMost``, and ``AtLeast``.
+These predicates enforce, respectively, that exactly, at most, or at least N of their ``BooleanVar`` arguments are ``True``.
+
+Usage:
+
+- :code:`AtLeast(3, Y[1], Y[2], Y[3])`
+- :code:`AtMost(3, Y)`
+- :code:`Exactly(3, Y)`
+
+.. doctest::
+
+    >>> m = ConcreteModel()
+    >>> m.my_set = RangeSet(4)
+    >>> m.Y = BooleanVar(m.my_set)
+    >>> m.p = LogicalConstraint(expr=AtLeast(3, m.Y))
+    >>> TransformationFactory('core.logical_to_linear').apply_to(m)
+    >>> m.logic_to_linear.pprint()  # constraint auto-generated by transformation
+    logic_to_linear : Size=1, Index=logic_to_linear_index, Active=True
+        Key : Lower : Body                                                          : Upper : Active
+          1 :   3.0 : Y_asbinary[1] + Y_asbinary[2] + Y_asbinary[3] + Y_asbinary[4] :  +Inf :   True
+    >>> m.p.pprint()
+    p : Size=1, Index=None, Active=False
+        Key  : Body                                 : Active
+        None : AtLeast(3: [Y[1], Y[2], Y[3], Y[4]]) :  False
+
+Indexed logical constraints
+---------------------------
+
+Like ``Constraint`` objects for numerical expressions, ``LogicalConstraint`` objects can be indexed.
+An example of this usage may be found below for the expression:
+
+.. math::
+
+    Y_{i+1} \Rightarrow Y_{i}, \quad i \in \{1, 2, \dots, n-1\}
+
+.. doctest::
+
+    >>> m = ConcreteModel()
+    >>> n = 5
+    >>> m.I = RangeSet(n)
+    >>> m.Y = BooleanVar(m.I)
+
+    >>> @m.LogicalConstraint(m.I)
+    ... def p(m, i):
+    ...     return m.Y[i+1] >> m.Y[i] if i < n else True
+
+    >>> m.p.pprint()
+    p : Size=4, Index=I, Active=True
+        Key : Body         : Active
+          1 : Y[2] >> Y[1] :   True
+          2 : Y[3] >> Y[2] :   True
+          3 : Y[4] >> Y[3] :   True
+          4 : Y[5] >> Y[4] :   True
+
+Integration with Disjunctions
+-----------------------------
+
+The logical expression system is designed to augment the previously introduced ``Disjunct`` and ``Disjunction`` components, the only original logical modeling types in Pyomo.GDP.
```

Comment:
I thought those got deprecated along with the `mpec` module? We still suppor them?

Reply:
Yeah, they're supported. `mpec` isn't deprecated.

Reply:
:open_mouth: I must have gotten it confused with `bilevel`. How would you word this then? Add `Complementarity` to the list? It's always felt a bit the odd child to me, since I didn't work much with them.

Reply:
Oh, whoops, I misread Pyomo.GDP as Pyomo. It's actually fine, because `Complementarity` is in `mpec`, not `gdp`. I think we can leave it--Sorry!